### ISSUE TITLE
`tsconfig.json` included as JSON input reports TS1006 self-reference

### ISSUE THREAD

**Issue by rubenferreira97** [microsoft/typescript-go#4275]
## Steps to reproduce

Repository: https://github.com/rubenferreira97/tsgo-tsconfig-self-reference-regression

The repro imports `tsconfig.json` from `index.ts`:

```ts
import tsconfig from "./tsconfig.json" with { type: "json" };

console.log(tsconfig);
```

And also includes `tsconfig.json` in the project:

```json
"include": ["index.ts", "tsconfig.json"]
```

## Behavior with `typescript@6.0`

`typescript@6.0.3` succeeds:

```powershell
npm run check:tsc
```

This runs:

```powershell
tsc --build
```

No diagnostics are reported.

## Behavior with `tsgo`

`@typescript/native-preview@7.0.0-dev.20260609.1` succeeds:

```powershell
npm run check:tsgo-working
```

`@typescript/native-preview@7.0.0-dev.20260610.1` fails:

```powershell
npm run check:tsgo-failing
```

Error:

```text
error TS1006: A file cannot have a reference to itself.
  The file is in the program because:
    Matched by include pattern 'tsconfig.json' in '.../tsconfig.json'
```

This appears to be a regression introduced between `7.0.0-dev.20260609.1` and `7.0.0-dev.20260610.1`.
